### ISSUE TITLE
Deleting project from DependencyTrack fails when deleting pod

### ISSUE THREAD

**Issue by muellerst-hg** [ckotzbauer/sbom-operator#183]
Deleting a pod results in error
```
time="2022-09-28T14:21:56Z" level=error msg="Could not load project: The project could not be found. (status: 404)"
```

In [`DependencyTrackTarget.Remove()`](https://github.com/ckotzbauer/sbom-operator/blob/main/internal/target/dtrack/dtrack_target.go#L173), populating `g.imageProjectMap` fails and remains empty after `g.LoadImages()`. As a result, dtrack project uuid for the image cannot be found. 

There is a bug in [`DependencyTrackTarget.LoadImages()`](https://github.com/ckotzbauer/sbom-operator/blob/main/internal/target/dtrack/dtrack_target.go#L173):
`imageId` is set to empty string at the beginning of project tag for loop. Therefore it will only be added to `imageProjectMap` if `raw-image-id` is the last tag in `project.Tags`. However, the `sbom-operator` usually follows `raw-image-id` and resets `imageId`
To fix this particular issue `imageId = ""` should be moved above the loop.

However, this is not enough to fix the actual problem that  deleting a project from DependencyTrack fails.
Its seems there are inconsistencies in the imageProjectMap. Once sbom-operator is running, new pods successfully create new dtrack project, but the imageId and corresponding uuid are not added to imageProjectMap. When the pod gets deleted, the project uuid resolved from imageProjectMap is 00000000-0000-0000-0000-000000000000 - the default for empty `uuid.UUID{}.String()` - which cannot be found in dtrack.
A potential fix could be to update `imageProjectMap` at the end of `ProcessSbom`.

**Comment by muellerst-hg** [comment]
I could send PRs for both issues if you like.

**Comment by ckotzbauer** [comment]
That would be great! @muellerst-hg

**Comment by ckotzbauer** [comment]
Much thanks for your two PRs and your research @muellerst-hg!!
I merged one of your PRs and rejected the other one. The reason for the rejection: I don't want to reload all projects on each SBOM processing and there was still no logic to maintain the `imageProjectMap` in the `ProcessSbom` and `Remove` function, so it was never up-to-date. I fixed that now (https://github.com/ckotzbauer/sbom-operator/commit/862539af99a6cf4b0bf656becebcf072cf188aaa). There will be a new release soon.

Thanks again!
